### PR TITLE
Add ORM base and date-based fetch suport.

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: #115579, tracked in MR with https://central.tri.be/issues/116356

This PR leverages the work done in https://github.com/moderntribe/tribe-common/pull/784 and previously done for Event Tickets to add the base of the ORM implementation for events and the first support of date-based filtering.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Event.php:76](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L76)
:x: | Each line in an array declaration must end in a comma | [Event.php:79](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L79)
:x: | Comma required after last value in array declaration | [Event.php:226](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L226)
:x: | Comma required after last value in array declaration | [Event.php:251](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L251)
:x: | String concat is not required here; use a single string instead | [Event.php:271](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L271)
:x: | String concat is not required here; use a single string instead | [Event.php:272](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L272)
:x: | String concat is not required here; use a single string instead | [Event.php:274](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L274)
:x: | String concat is not required here; use a single string instead | [Event.php:275](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L275)
:x: | Each line in an array declaration must end in a comma | [Event.php:366](https://github.com/moderntribe/the-events-calendar/blob/f4ed3700a0136ba1fd731e7103f0e49cbf96e6f9/src/Tribe/Repositories/Event.php#L366)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *Sprint 2*, but this was pull requested against *release/M18.15*.

File: src/Tribe/Query.php [hunk 1]
```diff
@@ -68,6 +68,11 @@ public static function can_inject_date_field( $query ) {
 		 * @param WP_Query $query
 		 */
 		public static function parse_query( $query ) {
+			// If this is set then the class will bail out of any filtering.
+			if ( $query->get( 'tribe_suppress_filters', false ) ) {
```
Comment (borkweb) [inline src/Tribe/Query.php]: Thought: Perhaps this should be `tribe_suppress_query_filters`, maybe? "Suppress filters" seems a bit too generic.

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M18.16*, but this was pull requested against *release/M18.15*.

**Comment by lucatume** [comment]
Fixed code issue, added changelog entry and created MR 18.16.

**Pull request merged**